### PR TITLE
Added macOS support.

### PR DESCRIPTION
**Pull request by firatagdas**

### CODE REVIEW

File: macos/.gitignore [hunk 1]
```diff
@@ -0,0 +1,31 @@
+.idea/
```
Comment (mogol) [inline macos/.gitignore:1]: Why do we need this file? Wouldn't project level enough?

File: macos/Classes/FlutterSecureStoragePlugin.m [hunk 2]
```diff
@@ -0,0 +1,201 @@
+#import "FlutterSecureStoragePlugin.h"
```
Comment (mogol) [inline macos/Classes/FlutterSecureStoragePlugin.m:1]: It seems pretty close (if not 100% same) to iOS version, can it be reused aka DRY?

Comment (talhabalaj) [reply macos/Classes/FlutterSecureStoragePlugin.m:1]: [package_info](https://github.com/flutter/plugins/blob/master/packages/package_info) implements an approach for making code dry between iOS and macOS.

Comment (firatagdas) [reply macos/Classes/FlutterSecureStoragePlugin.m:1]: well, i didn’t know that flutter plugins can handle symblinks. It is a good approach though

File: macos/.gitignore (hunk 1 above)
Comment (firatagdas) [reply macos/.gitignore:1]: Flutter adds automatically when you execute flutter create command to add missing platforms

File: macos/Classes/FlutterSecureStoragePlugin.m (hunk 2 above)
Comment (talhabalaj) [reply macos/Classes/FlutterSecureStoragePlugin.m:1]: yeah, I just saw it today.

**Comment by HeebeLee** [comment]
@mogol 
I need this, if you can, please merge as soon as possible. Thank you.

File: macos/Classes/FlutterSecureStoragePlugin.m (hunk 2 above)
Comment (HeebeLee) [reply macos/Classes/FlutterSecureStoragePlugin.m:1]: If the code is consistent with iOS, then this only requires the ln -s command. Of course, we can also share most of the code and do some meticulous work by judging the platform in OC.

I don't understand OC code, so please help to finish this work as soon as possible, thank you.

**Comment by muhlba91** [comment]
any progress on this so far? it's blocking us from porting apps to macos and would appreciate any release or pre-release with macos support.

**Comment by IonVillarreal** [comment]
I also have some applications that I would like to take to the Mac, there is something new about it ?

**Comment by jhancock4d** [comment]
Note that I've broken this out in PR #257 as a separate plugin with the platform interface implemented. Depending on the maintainer's wishes you may want to just work from that PR as it includes web and windows as well.

**Comment by juliansteenbakker** [comment]
I'm closing this because it has been integrated with #255 and #257

**Pull request closed without merging**